### ISSUE TITLE
[Bug] Various bugs...

### ISSUE THREAD

**Issue by Dark-Starr** [ProjectStarlight/StarlightRiver#179]
## Issue Descriptions
1- Missing Textures
	Expected behavior: Loaded Vanilla and SLR textures
	Actual behavior: Missing texture warnings, lots of them
	Context: Opening loot boxes, getting within range of stamina shards

2- Player's Rotation
	Expected behavior: Normal player appearance, not rotated at all
	Actual behavior: Player sprite rotated until reload
	Context: Gravity Orb lmao

3- Block Consumption
	Expected behavior: Blocks aren't eaten by the void when attempting to place them out of reach
	Actual behavior: Blocks are consumed by the abyss when attempting to place them out of reach
	Context: Building

4- Fargo's Compatibility
	Expected behavior: Easily accessible Codex and Stat sheet
	Actual behavior: Both tabs overlap, making the Codex inaccessible
	Context: I tried to open the Codex

5- Sentinel's Contrast..?
	Expected behavior: Blue-Green colour overlay when Barrier is active
	Actual behavior: Hyper-contrast orange overlay on player
	Context: Sentinel's Tincture

6- Instant Hook
	Expected behavior: Hook to blocks normally
	Actual behavior: Hooks deactivate the instant they hit a block
	Context: Grappling to ceilings, platforms

7- Lightweight Springs
	Expected behavior: Needing solid blocks to place a Hotspring Source
	Actual behavior: Midair Hotspring Source
	Context: Building again

8- Defying Gravity
	Expected behavior: I'm not sure
	Actual behavior: Hovering messes with Gravity Orbs
	Context: Gravity Orb again

## Reproduction
		Mod list: Better Boss Health Bar, Boss Checklist, Cheat Sheet, Fargo's Mutant Mod, Fargo's Testing Mod, Lights and Shadow, Magic Storage, Max Stack Plus Extra, Ore Excavator, Recipe Browser, Starlight River, WeaponDisplay
		Config settings: Default
		Game settings: Colour lighting, everything else default
		Gameplay events happening: Nothing major
		Your inputs: [1- Open loot box, load in/spawn Stamina Shard entity] [2- Spawn in Gravity Orb entity with Cheat Sheet]
		OS: Windows 10

## Additional context and log files
Images

1- Missing Textures
![image](https://user-images.githubusercontent.com/93879512/168180211-17e5ed6a-a9eb-4d36-8edf-4330a2bf4f8d.png)
![image](https://user-images.githubusercontent.com/93879512/168180248-2a4f62b6-224c-498d-9f9c-5208b4928416.png)
![image](https://user-images.githubusercontent.com/93879512/168180907-40f66148-462d-498e-83c9-1991f376648e.png)
2- Player's Rotation (First is rotated, second is normal)
![image](https://user-images.githubusercontent.com/93879512/168181272-493acb6b-ccfe-4fdd-a76c-a63161453c34.png)
![image](https://user-images.githubusercontent.com/93879512/168181408-220b329e-a6c6-4433-97fd-e137cf4c9b92.png)
3- Block Consumption
https://user-images.githubusercontent.com/93879512/168182003-1235a695-0e18-4c5b-ad64-6d4c7e158020.mp4
4- Fargo's Compatibility
![image](https://user-images.githubusercontent.com/93879512/168182360-137a11d0-3376-49bc-831a-1a7759f36edf.png)
5- Sentinel's Contrast..?
https://user-images.githubusercontent.com/93879512/168183592-a83e8446-bcd3-4cac-9eef-82f702edd246.mp4
6- Instant Hook
https://user-images.githubusercontent.com/93879512/168184237-c875c60c-905f-420d-97a0-5b42efbef6be.mp4
7- Lightweight Springs
![20220512194404_1](https://user-images.githubusercontent.com/93879512/168184874-ccff994e-d851-47bc-bf51-739cfc747edf.jpg)
8- Defying Gravity
https://user-images.githubusercontent.com/93879512/168187367-200ed4b8-d39c-4767-ba45-31912ee7dcf2.mp4

Logs (I hope this is the right file)
[client.log](https://github.com/ProjectStarlight/StarlightRiver/files/8683218/client.log)

**Comment by ScalarVector1** [comment]
1. partially addressed in d8ce8f8016a282fe2066187477ff8d330fcb7f94, need more detailed repro for the stamina shard issue.

2. fixed in d0c779e3acdb46453e317ac456dab8a0f18e0601

3. cannot reproduce. Please provide more repro details

4. wont fix.

5. Intended behavior with the solar flare armor, that is the pulsing effect from vanilla.

6. Appears to be related to unloading, only occurs after the mod has been reloaded atleast once. More research required

7. fixed in b416cf3ef1b2891b4dfb4a3ed2d1ec62e5db5a0e

8. Need more reproduction information and details. This report is far too vauge.

**Comment by Dark-Starr** [comment]
reproducing missing textures - stamina shard
spawned only entity with cheat sheet, not the debug tile

reproducing block consumption
really just try placing blocks in air, not connected to anything

side note on sentinel's contrast
even when not using armour or transformations, it does that

reproducing defying gravity
spawned gravity orb entity with cheat sheet
hold down/s while approaching it and while on it
hold jump to hover
and
jump off gravity orb
hover and fly left/right while being pulled
nyooooooom
tl;dr mess with hover for a bit, its really weird

**Comment by ScalarVector1** [comment]
Stamina shard issue safeguarded against with a5ac25853f9cd0aeab0a0b39eeb4dac799770c6f

Still cannot reproduce block consumption, please try to reproduce without maxStackPlus

Sentinel's contrast confirmed to be fault of Light and Shadows, wont fix

Gravity orb behavior is intended

**Comment by ScalarVector1** [comment]
Changing name as issue 6 here is the only reproducable and still relevant issue
